### PR TITLE
We can't use primary keys after all

### PR DESCRIPTION
**Pull request by simonrozsival**
## What's this?
We tried adding primary key attributes to the IDs of our Realm models so that we could use the `Find` method from Realm, but we realized it's not possible, because of the way we work with IDs of our entities: 
- an entity gets a temporary negative ID when it is created in the app
- after the app is pushed to the server, we get a final server-assigned positive ID and we change the entity ID from the negative one to the positive one
- Realm primary keys can't change. We would have to delete the old record and insert the new record (with the new ID). Even though this could work, it would cause a bunch of issues: we would have to rewire some dependencies upstream (e.g., when we delete and re-insert a client, we would have to fix all the projects which used this client and so on).

The simplest way to fix this is to go back to `All<T>.Where(x.Id == id).SingleOrDefault()` which is battle-tested and has worked just fine for us.

### Relationships
Partially reverts #7457 

## Why do we want this?
We want the app to work without rewriting too much of the existing codebase.

## How is it done?
We can't use the primary keys because we're modifying the `Id`s of entities after we get the server-assigned ID when they are created on the server.

## :squid: Permissions
🦑

### CODE REVIEW

**Comment by simonrozsival** [comment]
I'll test if we can change the values of `[Indexed]` properties.

File: Toggl.Storage.Realm/Queries/ProcessPullResultQuery.cs [hunk 1]
```diff
@@ -230,7 +227,8 @@ private void updateTimeEntry(ITimeEntry timeEntry, RealmTimeEntry dbTimeEntry, R
 
             dbTimeEntry.RealmTags.Clear();
             tagsIds
-                .Select(tagId => realm.Find<RealmTag>(tagId))
+                .Select(tagId => realm.GetById<RealmTag>(tagId))
+                .Where(tag => tag != null)
```
Comment (kornelijepetak) [inline Toggl.Storage.Realm/Queries/ProcessPullResultQuery.cs:231]: Shouldn't this fail if there are no tags, rather than just silently being ignored?

Comment (kornelijepetak) [reply Toggl.Storage.Realm/Queries/ProcessPullResultQuery.cs:231]: Or is this also the case of potentially locally deleted thing after the push?

Comment (simonrozsival) [reply Toggl.Storage.Realm/Queries/ProcessPullResultQuery.cs:231]: I am not sure which one is better. I noticed that in `RealmConstructors.cs` we would throw (we use `Single` to fetch the entity by ID) if the entity was not there. I'll remove the `Where`.

**Pull request merged**